### PR TITLE
refactor: remove OrderedColumnDesc from proto

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?
close https://github.com/singularity-data/risingwave/issues/3638

limitations:
- some remaining usage not cleaned https://github.com/singularity-data/risingwave/issues/3203
- namings need to be improved https://github.com/singularity-data/risingwave/issues/2145
- `OrderPair` (in common::sort_util) and `FieldOrder` (in frontend) is basically the same. can be merged

### CODE REVIEW

File: proto/plan_common.proto [hunk 1]
```diff
@@ -53,7 +48,7 @@ message CellBasedTableDesc {
   uint32 table_id = 1;
   repeated ColumnDesc columns = 2;
   // TODO: may refactor primary key representations
-  repeated OrderedColumnDesc order_key = 3;
+  repeated ColumnOrder order_key = 3;
   repeated uint32 pk_indices = 4;
```
Comment (BugenZhao) [inline proto/plan_common.proto:52]: We may also remove this field in this PR and use order key index for storage table, so #3572 can be resolved. 🤣

**Review by BugenZhao — COMMENTED** [review]
LGTM

File: src/frontend/src/optimizer/property/order.rs [hunk 2]
```diff
@@ -55,7 +55,7 @@ impl fmt::Display for Order {
     }
 }
 
-#[derive(Clone)]
+#[derive(Clone, PartialEq)]
 pub struct FieldOrder {
```
Comment (BugenZhao) [inline src/frontend/src/optimizer/property/order.rs:59]: Can we reuse `common::OrderPair` in the frontend? @st1page

File: src/stream/src/from_proto/batch_query.rs [hunk 3]
```diff
@@ -38,18 +39,26 @@ impl ExecutorBuilder for BatchQueryExecutorBuilder {
...
             .collect_vec();
-        let order_types = pk_descs.iter().map(|desc| desc.order).collect_vec();
 
         let column_descs = table_desc
             .columns
             .iter()
             .map(ColumnDesc::from)
             .collect_vec();
+        // TODO: remove this
+        let pk_descs = table_desc
```
Comment (BugenZhao) [inline src/stream/src/from_proto/batch_query.rs:54]: Could you please open another issue for removing `common`'s `OrderedColumnDesc`?

File: src/stream/src/from_proto/batch_query.rs [hunk 4]
```diff
@@ -59,9 +68,9 @@ impl ExecutorBuilder for BatchQueryExecutorBuilder {
 
         // Use indices based on full table instead of streaming executor output.
         let pk_indices = table_desc
-            .pk_indices
+            .order_key
```
Comment (BugenZhao) [inline src/stream/src/from_proto/batch_query.rs:71]: Could you please enable `index.slt.disabled` to test whether order key is correct?

**Review by BugenZhao — COMMENTED** [review]
Generally LGTM! May ask @st1page for designing. 🥵

File: src/frontend/src/optimizer/property/order.rs (hunk 2 above)
Comment (xxchan) [reply src/frontend/src/optimizer/property/order.rs:59]: TODO 🥵

File: src/stream/src/from_proto/batch_query.rs (hunk 3 above)
Comment (xxchan) [reply src/stream/src/from_proto/batch_query.rs:54]: https://github.com/singularity-data/risingwave/issues/3203

File: src/stream/src/from_proto/batch_query.rs (hunk 4 above)
Comment (xxchan) [reply src/stream/src/from_proto/batch_query.rs:71]: failed 🥸
image

**Comment by TennyZhuang** [comment]
@st1page PTAL

File: src/stream/src/from_proto/batch_query.rs (hunk 4 above)
Comment (BugenZhao) [reply src/stream/src/from_proto/batch_query.rs:71]: At least it does not panic. 🤣

**Comment by st1page** [comment]
> @st1page PTAL

agree with @BugenZhao

**Comment by xxchan** [comment]
agree with what? Approve? 🥵

**Pull request merged**